src/Symfony/Component/DependencyInjection/ParameterBag/EnvPlaceholderParameterBag.php

```diff
@@ -62,9 +62,17 @@ public function getEnvPlaceholders()
 
     /**
      * Merges the env placeholders of another EnvPlaceholderParameterBag.
+     *
+     * @param EnvPlaceholderParameterBag $bag
      */
     public function mergeEnvPlaceholders(self $bag)
     {
-        $this->envPlaceholders = array_merge_recursive($this->envPlaceholders, $bag->getEnvPlaceholders());
+        $newPlaceholders = $bag->getEnvPlaceholders();
+
+        foreach ($newPlaceholders as $key => $newEntries) {
```

Comment:
> needless use of recursive functions

not needless at all: the array is recursive (width two depth level)

> array_values($this->envPlaceholders[$placeholder])[0]

this would create a full copy of the entire array just to get the first element. foreach is the most efficient way to do so when not knowing the first key.

> don't see the point in an array that will always have the structure

the point is using key as a precomputed deduplication index, so that e.g. array_unique (slow compared to indexes) is not needed anymore.

> I don't think that assignment in a return statement

no pb for that change, go for it